src/test/java/com/google/devtools/build/lib/rules/cpp/Cpp20ModulesConfiguredTargetTest.java

```diff
@@ -0,0 +1,111 @@
+package com.google.devtools.build.lib.rules.cpp;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertThrows;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.devtools.build.lib.analysis.ConfiguredRuleClassProvider;
+import com.google.devtools.build.lib.analysis.util.AnalysisMock;
+import com.google.devtools.build.lib.analysis.util.BuildViewTestCase;
+import com.google.devtools.build.lib.analysis.util.DummyTestFragment;
+import com.google.devtools.build.lib.packages.util.Crosstool;
+import com.google.devtools.build.lib.testutil.TestRuleClassProvider;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class Cpp20ModulesConfiguredTargetTest extends BuildViewTestCase {
+
+  @Override
+  protected ConfiguredRuleClassProvider createRuleClassProvider() {
+    ConfiguredRuleClassProvider.Builder builder = new ConfiguredRuleClassProvider.Builder();
+    TestRuleClassProvider.addStandardRules(builder);
+    builder.addConfigurationFragment(DummyTestFragment.class);
+    return builder.addRuleDefinition(new TestRuleClassProvider.MakeVariableTesterRule()).build();
+  }
+
+  @Test
+  public void testCpp20ModulesConfigurationNoFlags() throws Exception {
+    ImmutableList<String> targetList = ImmutableList.of("//foo:lib", "//foo:bin", "//foo:test");
+    scratch.file(
+        "foo/BUILD",
+        """
+                cc_library(
+                    name = 'lib',
+                    module_interfaces = ["foo.cppm"],
+                )
+                cc_binary(
+                    name = 'bin',
+                    module_interfaces = ["foo.cppm"],
+                )
+                cc_test(
+                    name = 'test',
+                    module_interfaces = ["foo.cppm"],
+                )
+                """);
+    for(String targetName: targetList) {
+      AssertionError e = assertThrows(AssertionError.class, () -> getConfiguredTarget(targetName));
+      assertThat(e).hasMessageThat().contains("requires --experimental_cpp20_modules");
```

Comment:
I think that this assertion always passes - due to a weird unit test setup we have. Have you tried changing the message, does it fail?

The pattern to use is https://cs.opensource.google/bazel/bazel/+/master:src/test/java/com/google/devtools/build/lib/rules/cpp/CcCommonTest.java;l=111-113;drc=03ac7ae4e2c7743ab797c8179d428235c1a9d052
```
    reporter.removeHandler(failFastHandler);
    getConfiguredTarget("//a:a");
    assertContainsEvent("..."); // or assertNoEvents(); 
```